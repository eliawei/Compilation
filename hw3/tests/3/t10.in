// check cast and in precedence

int foo() {
  return 10b;
}

void main() {
    set[5..6] myset1;
    if (foo() in (myset1 - 10 - 4))
      print("success");
    if (5 > (int)myset1 + 10)
      print("OK");
    bool x = 5 + (int)myset1 in myset1;
}
