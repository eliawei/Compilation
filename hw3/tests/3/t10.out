---end scope---
---end scope---
---end scope---
---end scope---
myset1 SET 0
x BOOL 1
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
foo ()->INT 0
main ()->VOID 0
